samples/cpp/hello_reshape_ssd/main.cpp

```diff
@@ -2,214 +2,149 @@
+        size_t image_height = reader->height();
 
-        output_info->setPrecision(Precision::FP32);
-
-        auto dumpVec = [](const SizeVector& vec) -> std::string {
-            if (vec.empty())
-                return "[]";
-            std::stringstream oss;
-            oss << "[" << vec[0];
-            for (size_t i = 1; i < vec.size(); i++)
-                oss << "," << vec[i];
-            oss << "]";
-            return oss.str();
-        };
-        std::cout << "Resulting input shape = " << dumpVec(input_shape) << std::endl;
-        std::cout << "Resulting output shape = " << dumpVec(output_shape) << std::endl;
-        // -----------------------------------------------------------------------------------------------------
+        // reshape model to image size and batch size
+        ov::Shape tensor_shape = model->input().get_shape();
 
-        // --------------------------- Step 4. Loading a model to the device
-        // ------------------------------------------
-        ExecutableNetwork executable_network = ie.LoadNetwork(network, device_name);
-        // -----------------------------------------------------------------------------------------------------
+        tensor_shape[0] = batch_size;
+        tensor_shape[1] = image_channels;
+        tensor_shape[2] = image_height;
+        tensor_shape[3] = image_width;
 
-        // --------------------------- Step 5. Create an infer request
+        std::cout << "Reshape network to the image size = [" << image_height << "x" << image_width << "] "
+                  << "with batch = " << batch_size << std::endl;
+        model->reshape({{model->input().get_any_name(), tensor_shape}});
+        printInputAndOutputsInfo(*model);
+
+        // Step 4. Apply preprocessing
+        const ov::Layout tensor_layout{"NHWC"};
+
+        // clang-format off
+        model = PrePostProcessor(model).
+            // 1) InputInfo() with no args assumes a model has a single input
+            input(InputInfo().
+                // 2) Set input tensor information:
+                // - precision of tensor is supposed to be 'u8'
+                // - layout of data is 'NHWC'
+                // - set static spatial dimensions to input tensor to resize from
+                tensor(InputTensorInfo().
+                    set_element_type(ov::element::f32).
+                    set_spatial_static_shape(
+                        tensor_shape[ov::layout::height_idx(tensor_layout)],
+                        tensor_shape[ov::layout::width_idx(tensor_layout)]).
+                    set_layout(tensor_layout)).
+                // 3) Adding explicit preprocessing steps:
+                // - convert u8 to f32
+                // - convert layout to 'NCHW' (from 'NHWC' specified above at tensor layout)
+                preprocess(PreProcessSteps().
+                    convert_element_type(ov::element::f32). // WA for CPU plugin
+                    convert_layout("NCHW")). // WA for CPU plugin
```

Comment:
why do we need explicit preprocess steps here? Precisions and layout can be automatically matched and proper operations are inserted.